app/addons/documents/sidebar/sidebar.react.jsx

```diff
@@ -205,19 +212,23 @@ function (app, FauxtonAPI, React, Stores, Actions, Components, DocumentViews) {
 
   var DesignDoc = React.createClass({
 
-    createIndexList: function () {
-      var sidebarListTypes = FauxtonAPI.getExtensions('sidebar:list');
-
-      if (_.isEmpty(sidebarListTypes) ||
-        (_.has(sidebarListTypes[0], 'selector') && sidebarListTypes[0].selector !== 'views')) {
-        sidebarListTypes.unshift({
+    componentWillMount: function () {
+      this.sidebarListTypes = FauxtonAPI.getExtensions('sidebar:list');
```

Comment:
can we pass the sidebarListTypes as a prop, too? this would also make testing easier as we would not have to register extensions in order to test the functionality of the sidebar.